bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/ImageProcessingDescriptor.java

```diff
@@ -0,0 +1,19 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.swt.internal.image;
+
+import org.eclipse.swt.graphics.*;
+
+public interface ImageProcessingDescriptor {
```

Comment:
Isn't this more something like a pixel-transformation or color-transformation?

Reply:
That's true. I am not sure about this as the interface is intended to cover more (like SVG preprocessing), which would not be pixel transformation but rather a more general processing concept. Still, currently it is not, so the question is whether we should make the name more specific now and change it later on with a potential extension or directly indicate via the name that it is supposed to be extended for such a case.

Reply:
Difficult question. But since it's internal and not widely used we could make it specific now for the current use-cases and change it later, when there is demand. Maybe things will go different directions than expected and then we would have unnecessarily generic code.

> I am not sure about this as the interface is intended to cover more (like SVG preprocessing), which would not be pixel transformation

Wouldn't this then require different methods with different signatures if more than just one pixel is transformed at the time? This could then be a second interface with a helper to wrap a pixel-wise transformator?